Generate Client using OpenAPI spec, Update Setup Guide (#2)
ballerina-platform/module-ballerinax-hubspot.crm.object.companies: README.md

     --header 'content-type: application/x-www-form-urlencoded' \
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   - Windows

     ```bash
     curl --request POST ^
     --url https://api.hubapi.com/oauth/v1/token ^
     --header 'content-type: application/x-www-form-urlencoded' ^
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   This command will return the access token necessary for API calls.

   ```json
   {
     "token_type": "bearer",
     "refresh_token": "<Refresh Token>",
     "access_token": "<Access Token>",
     "expires_in": 1800
   }
   ```

5. Store the access token securely for use in your application.

## Quickstart

To use the `HubSpot CRM Companies Connector` connector in your Ballerina application, update the `.bal` file as follows:

### Step 1: Import the module

Import the `hubspot.crm.obj.companies` module.

````java
import ballerinax/hubspot.crm.obj.companies;
````

### Step 2: Instantiate a new connector

> 1. Create a `OAuth2RefreshTokenGrantConfig` with the obtained access token and initialize the connector with it.
> 
> ````ballerina
> configurable OAuth2RefreshTokenGrantConfig & readonly auth = ?;
> 
> final contact:Client contactClient = check new ({ auth });

Reviewer: ```suggestion
1. Create a `companies:OAuth2RefreshTokenGrantConfig` with the obtained access token and initialize the connector with it.

````ballerina
configurable companies:OAuth2RefreshTokenGrantConfig & readonly auth = ?;

final contact:Client hubSpotCrmCompanies = check new ({ auth });
```

   - Linux/macOS

     ```bash
     curl --request POST \
     --url https://api.hubapi.com/oauth/v1/token \
     --header 'content-type: application/x-www-form-urlencoded' \
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   - Windows

     ```bash
     curl --request POST ^
     --url https://api.hubapi.com/oauth/v1/token ^
     --header 'content-type: application/x-www-form-urlencoded' ^
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   This command will return the access token necessary for API calls.

   ```json
   {
     "token_type": "bearer",
     "refresh_token": "<Refresh Token>",
     "access_token": "<Access Token>",
     "expires_in": 1800
   }
   ```

5. Store the access token securely for use in your application.

## Quickstart

To use the `HubSpot CRM Companies Connector` connector in your Ballerina application, update the `.bal` file as follows:

### Step 1: Import the module

> Import the `hubspot.crm.obj.companies` module.

Reviewer 2: ```suggestion
Import the `ballerinax/hubspot.crm.obj.companies` module.
```

````java
import ballerinax/hubspot.crm.obj.companies;
````

### Step 2: Instantiate a new connector

1. Create a `companies:OAuth2RefreshTokenGrantConfig` with the obtained access token and initialize the connector with it.

````ballerina
configurable companies:OAuth2RefreshTokenGrantConfig & readonly auth = ?;

final companies:Client hubSpotCrmCompanies = check new ({ auth });
````

2. Create a Config.toml file and, configure the obtained credentials in the above steps as follows:

````toml
[auth]

> clientId = "<Client Id>"
> clientSecret =  "<Client Secret>"
> refreshToken = "<Refresh Token>"
> credentialBearer =  "POST_BODY_BEARER"

Reviewer 2: ```suggestion
clientId = "<Client Id>"
clientSecret = "<Client Secret>"
refreshToken = "<Refresh Token>"
credentialBearer = "POST_BODY_BEARER"
```

[![Trivy](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.companies/actions/workflows/trivy-scan.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.companies/actions/workflows/trivy-scan.yml)
[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.companies/actions/workflows/build-with-bal-test-graalvm.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.companies/actions/workflows/build-with-bal-test-graalvm.yml)
[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-hubspot.crm.object.companies.svg)](https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.object.companies/commits/master)

> [![GitHub Issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-library/module/hubspot.crm.companies.companies.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-library/labels/module%hubspot.crm.object.companies)

Reviewer: ```suggestion
[![GitHub Issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-library/module/hubspot.crm.object.companies.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-library/labels/module%hubspot.crm.object.companies)
```

   - Linux/macOS

     ```bash
     curl --request POST \
     --url https://api.hubapi.com/oauth/v1/token \
     --header 'content-type: application/x-www-form-urlencoded' \
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   - Windows

     ```bash
     curl --request POST ^
     --url https://api.hubapi.com/oauth/v1/token ^
     --header 'content-type: application/x-www-form-urlencoded' ^
     --data 'grant_type=authorization_code&code=<CODE>&redirect_uri=<YOUR_REDIRECT_URI>&client_id=<YOUR_CLIENT_ID>&client_secret=<YOUR_CLIENT_SECRET>'
     ```

   This command will return the access token necessary for API calls.

   ```json
   {
     "token_type": "bearer",
     "refresh_token": "<Refresh Token>",
     "access_token": "<Access Token>",
     "expires_in": 1800
   }
   ```

5. Store the access token securely for use in your application.

## Quickstart

To use the `HubSpot CRM Companies Connector` connector in your Ballerina application, update the `.bal` file as follows:

### Step 1: Import the module

> Import the `ballerinax/hubspot.crm.obj.companies` module.

Reviewer: ```suggestion
Import the `ballerinax/hubspot.crm.obj.companies` and `ballerina/oauth2` module.
```

````java

> import ballerinax/hubspot.crm.obj.companies;

Reviewer: ```suggestion
import ballerina/oauth2;
import ballerinax/hubspot.crm.obj.companies;
```

````

### Step 2: Instantiate a new connector

1. Create a `companies:OAuth2RefreshTokenGrantConfig` with the obtained access token and initialize the connector with it.

````ballerina

> configurable companies:OAuth2RefreshTokenGrantConfig & readonly auth = ?;

Reviewer: Shall we use this structure ?

https://github.com/ballerina-platform/module-ballerinax-hubspot.crm.properties?tab=readme-ov-file#step-2-instantiate-a-new-connector

final companies:Client hubSpotCrmCompanies = check new ({ auth });
````

2. Create a Config.toml file and, configure the obtained credentials in the above steps as follows:

````toml
[auth]
clientId = "<Client Id>"
clientSecret = "<Client Secret>"
refreshToken = "<Refresh Token>"

> credentialBearer = "POST_BODY_BEARER"

Reviewer: I think we can remove this from the configurations.
```suggestion
```
